### PR TITLE
hw3

### PR DESCRIPTION
**Pull request by ASmirnova28**

### CODE REVIEW

File: src/test/resources/hw3/prop.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+name=Roman
+password=Jdi1234
```
Comment (komyagin) [inline src/test/resources/hw3/prop.properties:2]: If we use a property file why don't we use it for the site URL?

File: src/test/java/hw3/pageobject_model/tests/ex2/PageObjectEx2Test.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        //9.Assert that for each checkbox there is an individual log row and value is corresponded to the
+        //status of checkbox
+        assertCheckboxes(diffElemPage);
+
+        //9.Assert that for radio button there is a log row and value is corresponded to the status of radio button
+        assertRadio(diffElemPage);
+
+        //9.Assert that for dropdown there is a log row and value is corresponded to the selected value
+        assertInDropdown(diffElemPage);
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/tests/ex2/PageObjectEx2Test.java]: Do you think it is an obvious behavior?

File: src/test/java/hw3/pageobject_model/steps/DiffElemPageSteps.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        for (WebElement logRadio : diffElemPage.logSelenRadio()) {
+            Assert.assertTrue(logRadio.isDisplayed());
+        }
+    }
+
+    public static void assertInDropdown(DiffElemPage diffElemPage) {
+        for (WebElement logDropdrown : diffElemPage.colourDropDown()) {
+            Assert.assertTrue(logDropdrown.isDisplayed());
+        }
+    }
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/steps/DiffElemPageSteps.java]: These methods have the same behavior why don't we move it to a separate method?

File: src/test/java/hw3/pageobject_model/steps/DiffElemPageSteps.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        diffElemPage.selectWaterCheckbox();
+    }
+
+    public static void selectWindCheckbox(DiffElemPage diffElemPage) {
+        diffElemPage.selectWindCheckbox();
+    }
+
+    public static void selectRadio(DiffElemPage diffElemPage) {
+        diffElemPage.selectRadioSelen();
+    }
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/steps/DiffElemPageSteps.java]: These methods have the same behavior why don't we move it to a separate method?

File: src/test/java/hw3/pageobject_model/pages/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        webDriver.navigate().to(URL);
+    }
+
+    public void assertTitle(String title) {
+        assertEquals(webDriver.getTitle(), title);
+    }
+
+    public void login() throws IOException {
+        InputStream input = new FileInputStream("src/test/resources/hw3/prop.properties");
+        Properties prop = new Properties();
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/pages/HomePage.java]: It's not a good idea to load properties in that place. And we should move the path of the property file to the string constant.

File: src/test/java/hw3/pageobject_model/pages/HomePage.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import java.io.InputStream;
+import java.util.List;
+import java.util.Properties;
+
+import static org.testng.Assert.assertEquals;
+
+public class HomePage {
+
+    protected WebDriver webDriver;
+    protected String URL = "https://jdi-testing.github.io/jdi-light/index.html";
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/pages/HomePage.java]: You're using the property file and you're placing the site URL here, why?

File: src/test/java/hw3/pageobject_model/pages/DiffElemPage.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        return getLogRows("Wind");
+    }
+
+    public List<WebElement> logSelenRadio() {
+        return getLogRows("metal");
+    }
+
+    public List<WebElement> colourDropDown() {
+        return getLogRows("Colors");
+    }
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/pages/DiffElemPage.java]: These methods have the same behavior why don't we move it to a separate method?

File: src/test/java/hw3/pageobject_model/pages/DiffElemPage.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        waterCheckbox.click();
+    }
+
+    public void selectWindCheckbox() {
+        windCheckbox.click();
+    }
+
+    public void selectRadioSelen() {
+        radioSelen.click();
+    }
```
Comment (komyagin) [inline src/test/java/hw3/pageobject_model/pages/DiffElemPage.java]: These methods have the same behavior why don't we move it to a separate method?

File: pom.xml [hunk 9]
```diff
@@ -53,11 +37,20 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>${bonigarcia.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-nop</artifactId>
+            <version>1.7.13</version>
```
Comment (komyagin) [inline pom.xml]: Why don't you use a variable for the version of slf4j? And where do you use slf4j?

File: pom.xml [hunk 10]
```diff
@@ -13,34 +13,18 @@
...
-            <id>Full</id>
-            <activation>
-                <activeByDefault>false</activeByDefault>
-            </activation>
-            <properties>
-                <suite.file>${project.basedir}/src/test/resources/hw1/FullTestSuite.xml</suite.file>
-            </properties>
-        </profile>
-        <profile>
-            <id>AddAndSub</id>
```
Comment (komyagin) [inline pom.xml]: Why did you delete previous profiles?

**Review by komyagin — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 9 above)
Comment (ASmirnova28) [reply pom.xml]: If there is no slf4j dependency in the pom.xml, it gives me a  SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder" warning, so I figured I should add it.

File: pom.xml (hunk 10 above)
Comment (ASmirnova28) [reply pom.xml]: I've finished making hw3 changes in hw2 branch without branching from master first, so I got confused on how it all will merge and probably I deleted them by mistake. Should I put them back?

File: src/test/java/hw3/utils/PropertiesReader.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+ /*    public void login() throws IOException {
+        InputStream input = new FileInputStream("src/test/resources/hw3/prop.properties");
+        Properties prop = new Properties();
+        prop.load(input);
+        dropDownMenu.click();
+        loginField.sendKeys(prop.getProperty("name"));
+        passwordField.sendKeys(prop.getProperty("password"));
+        enterButton.click();
+    }*/
```
Comment (komyagin) [inline src/test/java/hw3/utils/PropertiesReader.java]: What is it for?

**Review by komyagin — APPROVED** [review]
HW3 is accepted
